service/service.go

```diff
@@ -435,25 +435,25 @@ func (s *Service) GetSecret(ctx context.Context, req *pbsvc.UserRequest) (*pbsvc
 	secretLocker.RLock()
 	defer secretLocker.RUnlock()
 
-	retrievedSecret, err := getActiveSecretRow()
+	// check for any active secret
+	exists, err := hasActiveSecret()
 	if err != nil {
-		logger.Error(consts.GetSecret, consts.MsgErrGetActiveSecret, err.Error())
+		logger.Error(consts.GetSecret, consts.MsgErrLookUpActiveSecret, err.Error())
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	// if no active key in DB, create a new secret
-	if retrievedSecret == nil {
-		// insert new secret
+	// no active key was found in DB, create and insert new secret
+	if !exists {
 		if err := insertNewSecret(); err != nil {
```

Comment:
`if !err` should replace `if !exists`
call `deactivateSecret()` before `insertNewSecret` to handle cases that if we ever have more than 1 active secret.

Reply:
In this way, our codes are self-maintaining.

Reply:
discussed in slack, resolving